external-adapter/src/validator.ts

```diff
@@ -1,8 +1,14 @@
-const { AdapterError } = require('./errors')
-const { Requester } = require('./requester')
-const { logger } = require('./logger')
+import { AdapterError } from './errors'
+import { Requester } from './requester'
+import { logger } from './logger'
+
+export class Validator {
+  input: any
+  customParams: any
+  validated: any
+  error: any
+  errored: any
```

Comment:
We could actually type this as `AdapterErrorResponse`

Reply:
Not sure if I exactly follow on this, but I added it.

Reply:
If you take a look at our `AdapterError` there we construct a error response type that we return to the client:

```javascript
toJSONResponse() {
    const showDebugInfo = process.env.NODE_ENV === 'development' || process.env.DEBUG === 'true'
    const errorBasic = { name: this.name, message: this.message }
    const errorFull = { ...errorBasic, stack: this.stack, cause: this.cause }
    return {
      jobRunID: this.jobRunID,
      status: this.status,
      statusCode: this.statusCode,
      error: showDebugInfo ? errorFull : errorBasic,
    }
  }
```

My suggestion was just to type that, and then use it as `errored: AdapterErrorResponse` instead of `errored: any`.

Reply:
Ah I see now! Will add.